### ISSUE TITLE
Cannot use Kreait\Firebase\Http\Auth as Auth because the name is already in use 

### ISSUE THREAD

**Issue by mirahost** [kreait/firebase-php#120]
#### 1. Which version of `kreait/firebase-php` are you using?

3.2.0

#### 2. Which version of PHP are you using?

7

#### 3. What's the issue?

When getDatabase is called it throws the following error:
Compile Error - Cannot use Kreait\Firebase\Http\Auth as Auth because the name is already in use in DOCROOT/vendor/kreait/firebase-php/src/Firebase/Database.php on line 10

#### 4. Code that lead to the issue
```

        $serviceAccount = \Kreait\Firebase\ServiceAccount::fromArray($config);
        $firebase = (new \Kreait\Firebase\Factory)
            ->withServiceAccountAndApiKey($serviceAccount, $apiKey)
            ->withDatabaseUri('https://example.firebaseio.com')
            ->create();

        $database = $firebase->getDatabase();
```

Result:

Compile Error - Cannot use Kreait\Firebase\Http\Auth as Auth because the name is already in use in DOCROOT/vendor/kreait/firebase-php/src/Firebase/Database.php on line 10

**Comment by jeromegamez** [comment]
I can't reproduce this error, please make sure that you have no `use Kreait\Firebase\Http\Auth` or `use Kreait\Firebase\Http\Auth as Auth` statement elsewhere in your code.

**Comment by asahasrabuddhe** [comment]
@mirahost are you using this piece of code in conjunction with Laravel's Auth facade?

**Comment by renddrew** [comment]
I'm getting the same error in an otherwise empty project. Installed with composer.

**Comment by jeromegamez** [comment]
@mirahost @renddrew I still can't reproduce the issue, here is what I did:

```bash
mkdir fb-issue-120
cd fb-issue-120
composer require kreait/firebase-php
```

Then I created a `test.php` file:

```php
<?php

require __DIR__.'/vendor/autoload.php';

use Kreait\Firebase;

$serviceAccount = Firebase\ServiceAccount::fromJsonFile('/path/to/google-service-account.json');
$apiKey = '<Firebase Web API key>';

$firebase = (new Firebase\Factory)
    ->withServiceAccountAndApiKey($serviceAccount, $apiKey)
    ->create();

$database = $firebase->getDatabase();
```

The execution of `php test.php` then returned no error.

**Comment by ifilmstuff** [comment]
Same. @jeromegamez I started over with your response... still getting the issue "Cannot use Kreait\Firebase\Http\Auth as Auth"

**Comment by renddrew** [comment]
@jeromegamez I started fresh following your example and got the same error. I had started with a clear composer cache as well. I've been using your latest version where I renamed the Auth references in kreait/firebase-php/src/Firebase/Database.php. Not sure how legit this solution is but seems to be working for now.

**Comment by asahasrabuddhe** [comment]
Hello @mirahost @renddrew @ifilmstuff 

I am not able to reproduce this error either. Here is what I have done:

<?php

require __DIR__.'/vendor/autoload.php';

use Kreait\Firebase\ServiceAccount;
use Kreait\Firebase\Factory;

```
$serviceAccount = ServiceAccount::fromJsonFile('<PATH TO JSON FILE>');
$apiKey = '<API KEY>';

$firebase = (new Factory)
    ->withServiceAccountAndApiKey($serviceAccount, $apiKey)
    ->withDatabaseUri('https://<PROJECT ID>.firebaseio.com/')
    ->create();

$database = $firebase->getDatabase();
```
This code working perfectly without any error. I am using Composer 1.5.2 on PHP 7 on Ubuntu 16.04 system. I am still looking into this trying to figure out what might have gone wrong for you.

@renddrew If renaming the references is working for you, this has got something to do with the loading of classes which should not happen. Let me try something else in the meanwhile. Also, if you guys could just put the code you are working with on a repository and share it with me, I'd be able to look at what you have and try to find a solution from that angle. Thanks!

Regards,
Ajitem

**Comment by renddrew** [comment]
Hi @asahasrabuddhe here is a test build where I am seeing the error. https://github.com/renddrew/firebase-php-120

I am using MacOS 10.12.6 and Composer 1.5.2

**Comment by jeromegamez** [comment]
@renddrew @mirahost Please make sure that your HTTP server uses the same PHP runtime as your CLI/composer. You can check this by adding `echo phpversion();` after the `use` statement at the top of the test script.

I ensured that the library works as expected by using your repository with PHP 7.0.24 and PHP 7.1.0 inside docker:

```bash
git clone git@github.com:renddrew/firebase-php-120.git
cd firebase-php-120
cp /path/to/my/firebase_credentials.json firebase_credentials.json

# Test with PHP 7.0
docker run --rm -v $(pwd):/src -w /src php:7.0-alpine php test.php
# Test with PHP 7.1
docker run --rm -v $(pwd):/src -w /src php:7.1-alpine php test.php
```

**Comment by jeromegamez** [comment]
@mirahost @bennyt2 @renddrew @asahasrabuddhe 

Could you please try and use the version `dev-bugfix/120-auth` in your composer.json to see if this fixes the issue? In the composer.json it should look like this:

```json
"kreait/firebase-php": "dev-bugfix/120-auth"
```

Alternatively, you also can do 

```bash
composer require kreait/firebase-php:dev-bugfix/120-auth
```

I really liked the changes @bennyt2 did in #125, but I have to make sure that there are no backwards compatibility breaks, so it would be great if this change worked for you.

**Comment by jeromegamez** [comment]
Thank you for the feedback, @renddrew, I just released [v3.3.3](https://github.com/kreait/firebase-php/releases/tag/3.3.3) which includes the changes.

If the issue still remains with that release, please let me know - until then I will close this issue and hope for the best :)
